packages/ssz/src/core/clone.ts

```diff
@@ -62,10 +61,14 @@ function _clone(value: any, type: FullSSZType): any {
   const obj: any = {};
   switch (type.type) {
     case Type.uint:
-      if (BN.isBN(value)) {
-        return value.clone();
+      switch (type.use) {
+        case UintImpl.bn:
+          return value.clone();
+        case UintImpl.bigint:
+        case UintImpl.number:
+          return value;
       }
-      return value;
+      break;
```

Comment:
I think defaulting to bigint or trowing exception would be better solution

Reply:
ok, its meant to be unreachable code, will throw an error